src/routes/settings/components/Settings.js

```diff
@@ -0,0 +1,235 @@
+
+        this.state = {
+            firstName: props.userDetails.firstName || '',
+            lastName: props.userDetails.lastName || '',
+            email: props.userDetails.email || '',
+            currentPassword: '',
+            password: '',
+            confirmPassword: '',
+            isBlurred: {
+                email: false,
+                confirmPassword: false
+            }
+        };
+
+        this.strangeForm({
+            fields: ['firstName', 'lastName', 'email', 'currentPassword', 'password', 'confirmPassword'],
+            get: (someProps, field) => this.state[field],
+            act: (field, value) => this.setState({ [field]: value }),
+            getFormValue: {
+                '*': this.getFormValue.bind(this)
+            }
+        });
+
+        this.fieldBlurred = this._fieldBlurred.bind(this);
+        this.showEmailError = this._showEmailError.bind(this);
+        this.showPasswordError = this._showPasswordError.bind(this);
+        this.disableButton = this._disableButton.bind(this);
+        this.submit = this._submit.bind(this);
+    }
+
+    getCheckedValue(e) {
+
+        return e.target.checked;
+    }
+
+    getFormValue(e) {
+
+        return e.target.value || '';
+    }
+
+    _fieldBlurred(ev) {
+
+        const isBlurred = { ...this.state.isBlurred };
+        const field = ev.target.id;
+
+        isBlurred[field] = true;
+
+        this.setState({ isBlurred });
+    }
+
+    _showEmailError() {
+
+        return (this.state.isBlurred.email) && !IsEmail(this.state.email);
+    }
+
+    _showPasswordError() {
+
+        return this.state.isBlurred.confirmPassword && (this.state.password !== this.state.confirmPassword);
+    }
+
+    _disableButton() {
+
+        const { firstName, lastName, email, currentPassword, password, confirmPassword } = this.state;
+        // const fieldHasValue = (firstName && lastName && email && password) !== '';
+        const fieldHasChanged =
+            (firstName !== this.props.userDetails.firstName)
+            || (lastName !== this.props.userDetails.lastName)
+            || (email !== this.props.userDetails.email);
+        const passwordsMatch = password === confirmPassword;
+        const requestPasswordChange = (password !== '')
+            && passwordsMatch
+            && (currentPassword !== password)
+            && (currentPassword !== '');
+
+        // if (fieldHasValue && IsEmail(email) && passwordsMatch) {
+        //     return false;
+        // }
+
+        // TODO: Make sure this OR statement doesn't mess things up if something is half-filled or something
+        if (fieldHasChanged || (requestPasswordChange)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    _submit(ev) {
+
+        const { firstName, lastName, email, password } = this.state;
+
+        this.props.onSubmit({ email, password, firstName, lastName  });
+
+        ev.preventDefault();
+    }
+
+    render() {
+
+        return (
+
```

Comment:
I've started seeing this empty line in our components, and I don't know it's origin!